src/lib.rs

```diff
@@ -504,6 +534,8 @@ pub async fn run(config: BotConfig) -> anyhow::Result<()> {
             .context("writing new device_id into the database")?;
     }
 
+    let modules_config = config.modules_config.unwrap_or(HashMap::new());
```

Comment:
nit: Can you use `unwrap_or_else(HashMap::new)`  here? This should be better as it would only create the
hashmap if the value was None (current code creates it all the time).